### DISCUSSION TITLE
如何限制某个节点的移动

### DISCUSSION THREAD

**Discussion by zmcode** [antvis/X6#2217] [Q&A]
单数据请求回来的时候, 如何限制某个节点不可移动, 我看似乎可以通过translating来设置, 但是我在初始化完之后怎么改变这个配置呢?
![image](https://user-images.githubusercontent.com/48686959/173492917-1df396b4-76af-49c9-b876-06c5f54954d2.png)

**Comment by zmcode** [comment]
我看有一个node的方法可以改变移动, 但是,可以在我新增节点的时候, 就可以配置吗?
![image](https://user-images.githubusercontent.com/48686959/173500566-25e22193-89d3-44be-b108-df8258b6eb3e.png)

**Comment by zmcode** [comment]
![image](https://user-images.githubusercontent.com/48686959/173508477-30d11c4a-078a-4fd7-adf0-662b4dc731bf.png)
通过fromJSON生成数据, 然后拿到所有node, 拿到每个node实例, 想限制它的移动, 可是没有效果, 为啥? 能直接在创建node的时候, 就可以限制它的移动吗?

**Comment by zmcode** [comment]
https://x6.antv.vision/zh/docs/tutorial/epilog/#%E6%80%8E%E4%B9%88%E7%A6%81%E6%AD%A2%E8%8A%82%E7%82%B9%E7%A7%BB%E5%8A%A8
认真看了下文档可以通过interacting.nodeMovable来控制节点是否可以移动, 没有方法可以整体的重新配置graph实例的属性吗?

**Reply by NewByVector** [reply]
希望这个 [demo](https://codesandbox.io/s/dong-tai-shu-xing-4l1ld9) 能帮到你。

**Reply by itingkong** [reply]
按照demo设置节点是否可以移动，对动态添加的节点无效，依然可以移动

**Reply by itingkong** [reply]
使用 Graph.registerVueComponent() 注册的vue组件，并动态生成的节点

**Reply by NewByVector** [reply]
有 demo看下吗？
